test/extensions/filters/http/cache/simple_http_cache/simple_http_cache_test.cc

```diff
@@ -372,6 +373,71 @@ TEST_F(SimpleHttpCacheTest, UpdateHeadersDisabledForVaryHeaders) {
   EXPECT_TRUE(expectLookupSuccessWithHeaders(lookup(request_path_1).get(), response_headers_1));
 }
 
+TEST_F(SimpleHttpCacheTest, UpdateHeadersSkipEtagHeader) {
+  const std::string request_path_1("/name");
+  Http::TestResponseHeaderMapImpl response_headers_1{{"date", formatter_.fromTime(current_time_)},
+                                                     {"cache-control", "public,max-age=3600"},
+                                                     {"etag", "0000-0000"}};
+  insert(request_path_1, response_headers_1, "body");
+  EXPECT_TRUE(expectLookupSuccessWithHeaders(lookup(request_path_1).get(), response_headers_1));
+
+  // Update the date field in the headers
+  time_source_.advanceTimeWait(Seconds(3601));
+  std::string time_now = formatter_.fromTime(current_time_);
+  Http::TestResponseHeaderMapImpl response_headers_2{
+      {"date", time_now}, {"cache-control", "public,max-age=3600"}, {"etag", "1111-1111"}};
+  // The etag header should not be updated
+  Http::TestResponseHeaderMapImpl response_headers_3{
+      {"date", time_now}, {"cache-control", "public,max-age=3600"}, {"etag", "0000-0000"}};
+
+  updateHeaders(request_path_1, response_headers_2, {current_time_});
+
+  EXPECT_TRUE(expectLookupSuccessWithHeaders(lookup(request_path_1).get(), response_headers_3));
+}
+
+TEST_F(SimpleHttpCacheTest, UpdateHeadersSkipSpecificHeaders) {
+  const std::string request_path_1("/name");
+
+  // Vary not tested because we have separate tests that cover it
+  Http::TestResponseHeaderMapImpl response_headers_1{
+      {"date", formatter_.fromTime(current_time_)},
+      {"cache-control", "public,max-age=3600"},
+      {"content-range", "bytes 200-1000/67589"},
+      {"content-length", "800"},
+      {"content-type", "text/html; charset=UTF-8"},
+      {"etag", "0000-0000"},
+      {"link", "<https://example.com>; rel=\"preconnect\""}};
+  insert(request_path_1, response_headers_1, "body");
+  EXPECT_TRUE(expectLookupSuccessWithHeaders(lookup(request_path_1).get(), response_headers_1));
+  time_source_.advanceTimeWait(Seconds(100));
+
+  Http::TestResponseHeaderMapImpl response_headers_2{
+      {"date", formatter_.fromTime(current_time_)},
```

Comment:
put this formatted time in a temp rather than calling fromTime multiple times. In particular, it is not obvious whether current_time_ might advance between the calls, and temping it would put that issue to rest.

Reply:
Thanks for pointing this out!
```
  std::string str_1 = formatter_.fromTime(current_time_);
  time_source_.advanceTimeWait(Seconds(3601));
  std::string str_2 = formatter_.fromTime(current_time_);
```
This is a pattern that's used a lot in this test file. It's an incorrect use of the `time_source_` and `current_time_`. The `current_time_` doesn't get updated upon `time_source_.advanceTimeWait()`.  We only found out about this yesterday. I'll fix all occurrences of this pattern in this PR.